Update remote store documentation to reflect request level durability (#2320)
opensearch-project/documentation-website: _opensearch/remote.md

  "settings": {
    "index": {
      "number_of_shards": 1,
      "number_of_replicas": 1,
      "replication": {
        "type": "SEGMENT"
      },
      "remote_store": {
        "enabled": true,
        "repository": "segment-repo",

>         "translog": {

Reviewer: As `translog` is optional part in `remote_store` setting, we should document the difference between enabling it vs not.

Author: Yes, this makes sense. I will make the appropriate changes. Basically, the difference is when you have just the remote_store enabled, it would guarantee refresh level durability, but when translog is enabled as well, it would guarantee request level durability.

          "enabled": true,
          "repository": "translog-repo"
        }
      }
    }
  }
}
'
```

> The segment & translog repository can be same or different. All data that is added to the index will also be continuously uploaded to the remote storage on account of refreshes / flushes / translog fsync (to disk) in the form of segment and translog files. Along with this, there are other metadata files uploaded.

Reviewer 2: ```suggestion
The segment repository and translog repository can be the same or different. As data is added to the index, it will also be continuously uploaded to the remote storage in the form of segment and translog files because of refreshes, flushes, and translog fsync to disk. Along with data, other metadata files will be uploaded.
```

```

If the security plugin is enabled, a user must have the `cluster:admin/remotestore/restore` permission. See [Access control](/security-plugin/access-control/index/) for information about configuring user permissions.
{: .note}

## Potential use cases

> The following are potential use cases for the remote-backed storage feature:

Reviewer 2: ```suggestion
You can use remote-backed storage for the following purposes:
```

> - The ability to restore red clusters / indices.

Reviewer 2: ```suggestion
- To restore red clusters or indexes
```

> - The ability to recover all data until the last acknowledged write regardless of replica count if `index.translog.durability=request`.

Reviewer 2: ```suggestion
- To recover all data up to the last acknowledged write regardless of replica count if `index.translog.durability` is set to `request`
```

## Known limitations

The following are known limitations of the remote-backed storage feature:

> - Writing data to a remote store can be a high latency operation when compared to writing data on local file system. This, in expectation, impacts the indexing throughput and latency.

Reviewer 2: ```suggestion
- Writing data to a remote store can be a high latency operation when compared to writing data on local file system. This may impact the indexing throughput and latency.
```

> - Primary to primary relocation is unstable as handover of upload of translogs from older primary to new is yet to be implemented.

Reviewer 2: ```suggestion
- Primary to primary relocation is unstable because handover of translogs from the old primary to the new one is not implemented yet.
```

> - Garbage collection of metadata file is yet to be implemented.

Reviewer 2: ```suggestion
- Garbage collection of the metadata file is not implemented yet.
```

> - & [Other known issues](https://github.com/opensearch-project/OpenSearch/issues/5678).

Reviewer 2: ```suggestion
For other limitations, see the [Remote store known issue list](https://github.com/opensearch-project/OpenSearch/issues/5678).
```
